flang/lib/Lower/PFTBuilder.cpp

```diff
@@ -305,7 +306,10 @@ class PFTBuilder {
         // Link to the entry's first executable statement.
         entryPointList[entryIndex].second->lexicalSuccessor = p;
     } else if (const auto *entryStmt = p->getIf<parser::EntryStmt>()) {
-      const auto *sym = std::get<Fortran::parser::Name>(entryStmt->t).symbol;
+      const auto *sym = std::get<parser::Name>(entryStmt->t).symbol;
+      if (sym->IsFuncResult())
+        // Switch to the function sym.
+        sym = sym->owner().parent().FindSymbol(sym->name());
```

Comment:
This is a bad merge; these three lines should be deleted.

Reply:
Fixed.